### PR TITLE
msg: remove gencpp and genmsg submodules; remove gencpp imports

### PR DESCRIPTION
**Pull request by TSC21**
**Describe problem solved by this pull request**
After some verification, I noticed:
1. We are not using `gencpp` for anything (!) - there was probably a reason it was added some years ago, but now it's just there, occupying space as an unused dependency and submodule;
2. If we were using `genmsg` and `gencpp` from source because of lack of compatibility with other OS's, after some search I found that we do have genmsg available by pip: https://pypi.org/project/pyros-genmsg/. It just needs to be tested in other platforms, like Windows and MacOS.

**Describe your solution**
Completely remove the  `genmsg` and `gencpp` submodules and all the places where `gencpp` was being imported (since it wasn't really being used).

### CODE REVIEW

**Comment by dagar** [comment]
Sounds good to me. Goodbye submodules!

**Comment by TSC21** [comment]
For this to build correctly, we are going to need to update the base bionic and xenial containers to install `python-genmsg` by apt. `python3-genmsg` is being installed on the ROS2 containers, since is being used by `px4_ros_com` to generate the micro-RTPS agent code, but I guess to make this compatible with Python 3 as well, and following the developments in https://github.com/PX4/Firmware/pull/12809, the install of `python3-genmsg` should also be moved to the base containers.

**Comment by julianoes** [comment]
Tested on macOS, built fine after `pip3 install --user empy pyros-genmsg`.

**Comment by TSC21** [comment]
> Tested on macOS, built fine after `pip3 install --user empy pyros-genmsg`.

Great stuff! Let's make sure we update the containers accordingly! @MaEtUgR can you do something about AppVeyor?

**Comment by TSC21** [comment]
@dagar are you able to update the MacOS side on CI? For now, `pip2 install --user empy pyros-genmsg` should be enough (as we are still moving to Python3). But you can always install it for both and it stays already solved.

**Comment by TSC21** [comment]
Containers to be updated in https://github.com/PX4/containers/pull/213.

**Comment by TSC21** [comment]
I keep bumping into the same error as in :
```console
UnicodeDecodeError: 'ascii' codec can't decode byte 0xc2 in position 3356: ordinal not in range(128)
```
@julianoes do you know what might be causing this? I can't reproduce it locally (and I have the same Python version)

**Comment by TSC21** [comment]
We might have to setup [these](https://github.com/PX4/containers/blob/master/docker/Dockerfile_ros2-dashing#L10-L11) on the base containers. As soon as I get a chance I will test it locally on the containers.

**Comment by TSC21** [comment]
@MaEtUgR is it possible for you to update the Windows CI here for the python genmsg dependency?

**Comment by bkueng** [comment]
Can you improve the error outputs? I'm getting
```
python import error:  No module named 'genmsg'

Required python packages not installed.

On a Debian/Ubuntu system please run:

  sudo apt-get install python3-empy
  sudo pip3 install catkin_pkg

On MacOS please run:
  sudo pip3 install empy catkin_pkg

On Windows please run:
  easy_install empy catkin_pkg

```

Whereas `pyros-genmsg` is needed.

**Comment by TSC21** [comment]
> Can you improve the error outputs? I'm getting
> 
> ```
> python import error:  No module named 'genmsg'
> 
> Required python packages not installed.
> 
> On a Debian/Ubuntu system please run:
> 
>   sudo apt-get install python3-empy
>   sudo pip3 install catkin_pkg
> 
> On MacOS please run:
>   sudo pip3 install empy catkin_pkg
> 
> On Windows please run:
>   easy_install empy catkin_pkg
> ```
> 
> Whereas `pyros-genmsg` is needed.

Yes absolutely.

**Comment by TSC21** [comment]
https://github.com/PX4/containers/pull/214 brings the fix for the decoding issues. Tested locally and on Jenkins. Just needs a container update now.

**Comment by TSC21** [comment]
@dagar with the [update to GCC 9](https://github.com/PX4/containers/commit/25844a3e33d073764608ffcdeafe35960800fe57) on the containers, it's becoming a nightmare to get this in. Is there a follow-up work after the update?

**Comment by TSC21** [comment]
@MaEtUgR can you please bring `pyros-genmsg` as well to the Cygwin Toolchain? Thanks!

**Comment by MaEtUgR** [comment]
Looks like a good plan you have with this pr!
I added them to the Windows toolchain here: https://github.com/PX4/windows-toolchain/commit/8f8fbb42052a3718429534a7ee8f2965277dee52
I'll do a new toolcahin version and add it to CI asap.

**Comment by TSC21** [comment]
@MaEtUgR thanks for the toolchain update. The main issue now is with GCC 9.

**Comment by julianoes** [comment]
I need to make sure all dockerhub docker builds are ready and then re-run CI here.

**Comment by julianoes** [comment]
@dogmaphobic this fails on your Mac. Could you please install the pip dependencies for Python3?
```
pip3 install pyserial empy toml numpy pandas jinja2 pyyaml
```

**Comment by dogmaphobic** [comment]
> Could you please install the pip dependencies for Python3?
> 
> pip3 install pyserial empy toml numpy pandas jinja2 pyyaml

Done!

**Comment by dogmaphobic** [comment]
```
$ pip3 install pyserial empy toml numpy pandas jinja2 pyyaml
Collecting pyserial
Collecting empy
Collecting toml
Collecting numpy
Collecting pandas
Collecting jinja2
Collecting pyyaml
Collecting python-dateutil>=2.6.1
Collecting pytz>=2017.2
Collecting MarkupSafe>=0.23
Requirement already satisfied: six>=1.5 in /usr/local/lib/python3.7/site-packages (from python-dateutil>=2.6.1->pandas) (1.13.0)
Installing collected packages: pyserial, empy, toml, numpy, python-dateutil, pytz, pandas, MarkupSafe, jinja2, pyyaml
Successfully installed MarkupSafe-1.1.1 empy-3.3.4 jinja2-2.10.3 numpy-1.18.0 pandas-0.25.3 pyserial-3.4 python-dateutil-2.8.1 pytz-2019.3 pyyaml-5.2 toml-0.10.0
```

```
$ pip3 install empy pyros-genmsg
Requirement already satisfied: empy in /usr/local/lib/python3.7/site-packages (3.3.4)
Collecting pyros-genmsg
  Downloading https://files.pythonhosted.org/packages/37/46/c5e3076ce9480d4002e87268c09270a1db17d3173a7b97fe1501827d27c6/pyros_genmsg-0.5.8-py2.py3-none-any.whl
Installing collected packages: pyros-genmsg
Successfully installed pyros-genmsg-0.5.8
```

Still failing.
Any clues?

File: .ci/Jenkinsfile-hardware [hunk 1]
```diff
@@ -346,6 +346,110 @@ pipeline {
                     sh './Tools/HIL/run_tests.py --device `find /dev/serial -name *usb-FTDI_*`'
                   }
                 }
+                stage("stop modules") {
```
Comment (dagar) [inline .ci/Jenkinsfile-hardware]: Bad rebase?

Comment (TSC21) [reply .ci/Jenkinsfile-hardware]: Yes

**Comment by julianoes** [comment]
Some dependencies seem wrong because pip2 is overwritten by pip3 in some containers, see: https://github.com/PX4/containers/pull/222

**Comment by TSC21** [comment]
> Some dependencies seem wrong because pip2 is overwritten by pip3 in some containers, see: [PX4/containers#222](https://github.com/PX4/containers/pull/222)

Yeah let's hope that fixes it.

**Comment by TSC21** [comment]
@julianoes with respect to http://ci.px4.io:8080/blue/organizations/jenkins/PX4%2FFirmware/detail/PR-13572/33/pipeline, we need to add a condition in CMake that in the case of building with `catkin`, it looks for Python 2 instead of 3.

**Comment by TSC21** [comment]
WRT http://ci.px4.io:8080/blue/organizations/jenkins/PX4_misc%2FFirmware-SITL_tests/detail/PR-13572/33/pipeline#step-177-log-165: https://github.com/PX4/containers/pull/233.

**Comment by TSC21** [comment]
@LorenzMeier anything that can be done WRT http://ci.px4.io:8080/blue/organizations/jenkins/PX4_misc%2FFirmware-compile/detail/PR-13572/33/pipeline? Can you do it yourself or someone else with access to these? Thanks!

**Comment by TSC21** [comment]
@julianoes success!! Finally :D

File: msg/tools/px_generate_uorb_topic_files.py [hunk 2]
```diff
@@ -60,14 +55,14 @@
 
 On a Debian/Ubuntu system please run:
 
-  sudo apt-get install python-empy
-  sudo pip install catkin_pkg
+  sudo apt-get install python3-empy
+  sudo pip3 install pyros-genmsg
 
 On MacOS please run:
-  sudo pip install empy catkin_pkg
```
Comment (LorenzMeier) [inline msg/tools/px_generate_uorb_topic_files.py]: Have the docs been updated?

Comment (julianoes) [reply msg/tools/px_generate_uorb_topic_files.py]: Good point. We'll check them.

**Comment by TSC21** [comment]
Let's bring this in and update the docs where needed.

**Pull request merged**